kernel/packages/unity-interface/dcl.ts

```diff
@@ -116,6 +116,35 @@ const positionEvent = {
   mousePosition: Vector3.Zero()
 }
 
+/////////////////////////////////// AUDIO STREAMING ///////////////////////////////////
+
+const audioStreamSource = new Audio()
+
+teleportObservable.add(() => {
+  audioStreamSource.pause()
```

Comment:
As the audio is not spatialized, it wouldn't be more wise to listen to the player's position and stop it when the player is out of the scene? if not we are going to listen to the stream from very far away and it can be an unpleasant experience.

What happens if we teleport to coordinates inside the `AudioStream` range, it first pauses and then plays as intended?

What happens if we have 2 `AudioStream` components from 2 different scenes? They wouldn't override each other?

Reply:
> As the audio is not spatialized, it wouldn't be more wise to listen to the player's position and stop it when the player is out of the scene? if not we are going to listen to the stream from very far away and it can be an unpleasant experience.

This is how it actually works. Audio can only be played if player is inside the same scene as the AudioStream component.
That subscription to the teleport event is just to make sure that, no matter what, the audio stops when player teleports.

> What happens if we teleport to coordinates inside the AudioStream range, it first pauses and then plays as intended?

Yes

> What happens if we have 2 AudioStream components from 2 different scenes? They wouldn't override each other?

As mentioned before, the audio is only played if the user is in the same scene.. if player move out of scene audio stops, if player move to a scene with other AudioStream, the audio of that scene is played

Reply:
Checks if audio can or cannot be played are on the unity side of the component in `UpdatePlayingState` method of `DCLAudioStream` class